rpxy-bin/src/log.rs

```diff
@@ -35,8 +39,12 @@ fn init_file_logger(level: tracing::Level, log_dir_path: &str) {
   println!("Access log: {}", access_log_path.display());
   println!("System and error log: {}", system_log_path.display());
 
-  let access_log = open_log_file(&access_log_path);
-  let system_log = open_log_file(&system_log_path);
+  let access_log_appender = tracing_appender::rolling::daily(&log_dir_path, ACCESS_LOG_FILE);
+  let (access_non_blocking, guard) = tracing_appender::non_blocking(access_log_appender);
+  _ = ACCESS_LOG_GUARD.set(guard);
+  let system_log_appender = tracing_appender::rolling::daily(log_dir_path, SYSTEM_LOG_FILE);
+  let (system_non_blocking, guard) = tracing_appender::non_blocking(system_log_appender);
+  _ = SYSTEM_LOG_GUARD.set(guard);
```

Comment:
Here I see it is reasonable to use non_blocking IO. But I found an issue that this change using `tracing_appender::rolling::daily` races the log rotation in our container packages.

We use `logrotate` for logging file rotation based on file size in docker container.

https://github.com/junkurihara/rust-rpxy/blob/7363241c48e05422fa03ac67964465a10a6a91ec/docker/entrypoint.sh#L19-L61

Reply:
I believe your PR is really valuable since I didn't care about the file-lock behavior. Thanks for this PR! Let us continuously consider 'non-locking' way of log files based on this PR.

Reply:
It is possible to limit the number of files kept using `tracing-appender` (using the builder), however it does not support compression.
It would be possible to allow the user to configure the number of days kept and/or if they wanted the per day (or hour, or minute) log rotation to be enabled since the only rotation options in `tracing-appender` are minutely, hourly, daily, or never.
The file names can be adjusted within `tracing-appender`, however they will always contain the UTC date/time.

If you'd like these options, or a subset, I'd be open to working on adding support for them in the command line configuration.